plugins/basicstyles/plugin.js

```diff
@@ -101,6 +101,41 @@ CKEDITOR.plugins.add( 'basicstyles', {
 			[ CKEDITOR.CTRL + 73 /*I*/, 'italic' ],
 			[ CKEDITOR.CTRL + 85 /*U*/, 'underline' ]
 		] );
+	},
+
+	afterInit: function( editor ) {
+		var subscriptCommand = editor.getCommand( 'subscript' ),
+			superscriptCommand = editor.getCommand( 'superscript' ),
+			disallowAddSubSup = editor.config.coreStyles_disallowSubscriptSuperscript;
+
+		// Prevent adding subscript and superscript only when both buttons exists. (#5215)
+		if ( disallowAddSubSup && ( subscriptCommand && superscriptCommand ) ) {
+			editor.on( 'beforeCommandExec', function( evt ) {
+				if ( evt.data.name === 'subscript' ) {
+					// In case when superscript is active, we should remove it first.
+					if ( superscriptCommand.state === CKEDITOR.TRISTATE_ON ) {
+						editor.execCommand( 'superscript' );
+						// Lock snapshot to prevent adding additional undo step after removing superscript.
+						editor.fire( 'lockSnapshot' );
+					}
+				}
+
+				if ( evt.data.name === 'superscript' ) {
+					if ( subscriptCommand.state === CKEDITOR.TRISTATE_ON ) {
+						editor.execCommand( 'subscript' );
+						editor.fire( 'lockSnapshot' );
+					}
+				}
```

Comment:
Those two blocks seems to be identical, could we extract them to a single function with name adjusted to the abstraction level of this code, so:
```suggestion
				if ( evt.data.name === 'subscript' ) {
					offActiveCommand( superscriptCommand );
				}

				if ( evt.data.name === 'superscript' ) {
					offActiveCommand( subscriptCommand );
				}
```

and the function itself (improved based on [API](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_command.html#method-exec)):
```js
function offActiveCommand( command ){
				if ( command.state === CKEDITOR.TRISTATE_ON ) {
					command.exec( editor );
					editor.fire( 'lockSnapshot' );
				}
			}
```